src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -345,6 +348,54 @@ default Map<ChangeMode, Class<?>[]> getAcceptedChangeModes() {
 	 */
 	void change(Event event, @Nullable Object[] delta, ChangeMode mode);
 
+	/**
+	 * Changes the contents of an expression using the given {@link Function}.
+	 * Intended for changes that change properties of the values of the expression, rather than completely
+	 * changing the expression. For example, {@code set vector length of {_v} to 1}, rather than
+	 * {@code set {_v} to vector(0,1,0)}.
+	 * <br>
+	 * This is a 1 to 1 transformation and should not add or remove elements.
+	 * For {@link Variable}s, this will retain indices. For non-{@link Variable}s, it will
+	 * evaluate {@link #getArray(Event)}, apply the change function on each, and call
+	 * {@link #change(Event, Object[], ChangeMode)} with the modified values and {@link ChangeMode#SET}.
+	 *
+	 * @param event The event to use for local variables and evaluation
+	 * @param changeFunction A 1-to-1 function that transforms a single input to a single output.
+	 * @param <R> The output type of the change function. Must be a type returned
+	 *              by {{@link #acceptChange(ChangeMode)}} for {@link ChangeMode#SET}.
+	 */
+	default <R> void changeInPlace(Event event, Function<T, R> changeFunction) {
+		changeInPlace(event, changeFunction, false);
+	}
+
+	/**
+	 * Changes the contents of an expression using the given {@link Function}.
+	 * Intended for changes that change properties of the values of the expression, rather than completely
+	 * changing the expression. For example, {@code set vector length of {_v} to 1}, rather than
+	 * {@code set {_v} to vector(0,1,0)}.
+	 * <br>
+	 * This is a 1 to 1 transformation and should not add or remove elements.
+	 * For {@link Variable}s, this will retain indices. For non-{@link Variable}s, it will
+	 * evaluate the expression, apply the change function on each value, and call
+	 * {@link #change(Event, Object[], ChangeMode)} with the modified values and {@link ChangeMode#SET}.
+	 *
+	 * @param event The event to use for local variables and evaluation
+	 * @param changeFunction A 1-to-1 function that transforms a single input to a single output.
+	 * @param all Whether to evaluate with {@link #getAll(Event)} or {@link #getArray(Event)}.
+	 * @param <R> The output type of the change function. Must be a type returned
+	 *              by {{@link #acceptChange(ChangeMode)}} for {@link ChangeMode#SET}.
+	 */
+	default <R> void changeInPlace(Event event, Function<T, R> changeFunction, boolean all) {
```

Comment:
`useAll` might better clarify that it uses getAll instead of confusing this for whether to apply it to all values or something, but idk